src/main/java/com/moneylendingapp/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.moneylendingapp.services.impl;
+
+import com.moneylendingapp.dto.responses.SignUpResponse;
+import com.moneylendingapp.exceptions.BadRequestException;
+import com.moneylendingapp.dto.requests.SignUpRequest;
+import com.moneylendingapp.entities.User;
+import com.moneylendingapp.enums.EmploymentStatus;
+import com.moneylendingapp.repositories.UserRepository;
+import com.moneylendingapp.services.UserService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepo;
+    private PasswordEncoder passwordEncoder;
+
+    @Override
+    public SignUpResponse createUser(SignUpRequest signUpDto) {
+
+        validateUsername(signUpDto.getUsername());
```

Comment:
this is ambiguous.
maybe you can use ensureUserNameisUnique() which is more specific as to the kind of validation that is being done